Architecture_Engine/Query/OutlineElements1D.cs

```diff
@@ -57,6 +63,12 @@ public static List<IElement1D> OutlineElements1D(this Room room)
         [Output("outlineElements", "A collection of outline 1D elements")]
         public static List<IElement1D> OutlineElements1D(this Ceiling ceiling)
         {
+            if(ceiling == null)
+            {
+                BH.Engine.Reflection.Compute.RecordError("Cannot query the outline 1D elements of a null ceiling.");
+                return null;
```

Comment:
Why null and not an empty list?
```suggestion
                return new List<IElement1D>();
```